staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/helpers.go

```diff
@@ -232,3 +232,14 @@ func HasObjectMetaSystemFieldValues(meta Object) bool {
 	return !meta.GetCreationTimestamp().Time.IsZero() ||
 		len(meta.GetUID()) != 0
 }
+
+// ResetObjectMetaForStatus forces the meta fields for a status update to match the meta fields
+// for a pre-existing object.  Status updates are for updating status, not annotations
+func ResetObjectMetaForStatus(meta, existingMeta Object) {
+	meta.SetGeneration(existingMeta.GetGeneration())
+	meta.SetSelfLink(existingMeta.GetSelfLink())
+	meta.SetLabels(existingMeta.GetLabels())
```

Comment:
Same comment as for annotations, but less emphatically. Until we have a decent, working general field mechanism I think it probably does make sense to allow label and status updates at the same time. (use case: allow pod's node selector or affinity / taint selectors to match status-y things about nodes)

I don't know of anyone doing this currently, though?